tensorflow/core/common_runtime/process_util.cc

```diff
@@ -128,6 +128,9 @@ int32 NumInterOpThreadsFromSessionOptions(const SessionOptions& options) {
   const int32_t env_inter_op = GetEnvNumInterOpThreads();
   if (env_inter_op > 0) return env_inter_op;
 
+  // Reset inter to 1 since negative value means running in caller thread.
+  if (inter_op < 0 || env_inter_op < 0) return 1;
```

Comment:
This is potentially changing behavior, since we sometimes assume a zero or negative will instead create `MaxParallelism` threads.  We have calls to this function internally at Google that aren't from `direct_session.cc`.

Reply:
I have addressed code to accept a new arg, but I still have a question here: `run_in_call_thread_` is set if there's negative config value: https://github.com/tensorflow/tensorflow/blob/r2.9/tensorflow/core/common_runtime/direct_session.cc#L344-L348. but you said negative value will be used as different meaning in internal usage, will the internal usage break the design of `run_in_call_thread_`?

Reply:
It seems only `direct_session.cc` has the assumption that a negative value means `run_in_call_thread_`.  The `NumInterOpThreadsFromSessionOptions()` function itself, in `process_util.cc`, returns the max-parallelism value for negative inputs.

Reply:
Got it, thanks.